test/NuGet.Core.FuncTests/NuGet.Packaging.CrossVerify.Generate.Test/GenerateSignedPackages.cs

```diff
@@ -361,6 +361,42 @@ public async Task PreGenerateSignedPackages_AuthorSigned_Timestamped_RepositoryC
 
             }
         }
+
+#if IS_DESKTOP
+        [Fact]
+        public async Task PreGenerateSignedPackages_AuthorSigned_TimeStampedWithNoSigningCertificateUsage()
+        {
+            // Arrange
+            var folder = FolderNames.One.ToString();
+            string caseFolder = Path.Combine(_dir, folder);
+            Directory.CreateDirectory(caseFolder);
+
+            ISigningTestServer testServer = await _signingTestFixture_Author.GetSigningTestServerAsync();
+            CertificateAuthority rootCa = await _signingTestFixture_Author.GetDefaultTrustedCertificateAuthorityAsync();
+            var options = new TimestampServiceOptions()
+            {
+                SigningCertificateUsage = SigningCertificateUsage.None
+            };
+            TimestampService timestampService = TimestampService.Create(rootCa, options);
+
+            using (testServer.RegisterResponder(timestampService))
+            {
+                var nupkg = new SimpleTestPackageContext();
+
+                string packagePath = Path.Combine(caseFolder, "package");
+                Directory.CreateDirectory(packagePath);
+
+                using (var certificate = new X509Certificate2(_signingTestFixture_Author.TrustedTestCertificate.Source.Cert))
+                {
+                    string signedPackagePath = await SignedArchiveTestUtility.AuthorSignPackageAsync(
+                        certificate,
+                        nupkg,
+                        packagePath,
+                        timestampService.Url);
+                }
+            }
+        }
+#endif
     }
 }
 #endif
```

Comment:
perhaps the #endifs should have something like:
`#endif  //IS_SIGNING_SUPPORTED`

i just had to go search to see if you accidentally had 2 closes.